python_modules/libraries/dagster-snowflake/dagster_snowflake/resources.py

```diff
@@ -64,64 +275,76 @@ def __init__(self, config: Mapping[str, str], log):
             ),
         )
 
-        if self.connector == "sqlalchemy":
-            self.conn_args: Dict[str, Any] = {
-                k: config.get(k)
-                for k in (
-                    "account",
-                    "user",
-                    "password",
-                    "database",
-                    "schema",
-                    "role",
-                    "warehouse",
-                    "cache_column_metadata",
-                    "numpy",
-                )
-                if config.get(k) is not None
-            }
-
-            if (
-                config.get("private_key", None) is not None
-                or config.get("private_key_path", None) is not None
-            ):
-                # sqlalchemy passes private key args separately, so store them in a new dict
-                self.sqlalchemy_engine_args["private_key"] = self.__snowflake_private_key(config)
-
-        else:
-            self.conn_args = {
-                k: config.get(k)
-                for k in (
-                    "account",
-                    "user",
-                    "password",
-                    "database",
-                    "schema",
-                    "role",
-                    "warehouse",
-                    "autocommit",
-                    "client_prefetch_threads",
-                    "client_session_keep_alive",
-                    "login_timeout",
-                    "network_timeout",
-                    "ocsp_response_cache_filename",
-                    "validate_default_parameters",
-                    "paramstyle",
-                    "timezone",
-                    "authenticator",
-                )
-                if config.get(k) is not None
-            }
-            if (
-                config.get("private_key", None) is not None
-                or config.get("private_key_path", None) is not None
-            ):
-                self.conn_args["private_key"] = self.__snowflake_private_key(config)
-
-        self.autocommit = self.conn_args.get("autocommit", False)
-        self.log = log
-
-    def __snowflake_private_key(self, config) -> bytes:
+        return v
+
+    @property
+    @cached_method
```

Comment:
alternate implementation - we factor out `_connection_args` `_sqlalchemy_connection_args` `_sqlalcehmy_engine_args` and `get_connection` to a third class and then wrap the `get_connection` call in the `SnowflakeResource` and `SnowflakeConnection`